Updated the base image and the git-lfs installation method

LCerebo opened the pull request:
Updated the base image to the latest LTS ubuntu version.
Uniformed the git and git-lfs installation using the defaults apt repositories.

FulvioRizzardiniEng committed:
Update the base image to the latest lts ubuntu version.
Uniformed the git and git-lfs installation using the defaults apt repositories.

FulvioRizzardiniEng committed:
bugfix: delete the user if already exist in the container.
If the user uid already exist the useradd command fails, now if the user exist first it will be deleted and then recreated with the jenkins parameters.

FulvioRizzardiniEng committed:
Updated java version to the newest supported one.

shamil commented on Dockerfile:
```diff
@@ -1,29 +1,24 @@
     # Jenkins is run with user `jenkins`, uid = 1000
     # If you bind mount a volume from the host or a data container,
     # ensure you use the same uid
-    && groupadd -g ${gid} ${group} \
+    && if id -u ${uid}; then userdel $(id -n -u ${uid}) ; fi \
+    && groupadd -f -g ${gid} ${group} \
     && useradd -d "$JENKINS_HOME" -u ${uid} -g ${gid} -m -s /bin/bash ${user} \
```
Starting with Ubuntu 23.04, the Ubuntu devs. introduced a non-root default OCI user (`ubuntu`), along with a corresponding group, which was not present in earlier versions. This change is being discussed in [this Launchpad bug report](https://bugs.launchpad.net/cloud-images/+bug/2005129).

Your code will break the `groupadd` command because the `ubuntu` group already exists. As a result, the `groupadd` command will create a new group with a different `gid` than the one specified in `${gid}`.

To avoid this issue, we can remove both the `ubuntu` user and group. This will reset the environment to how it worked before 23.04, with only system users present. This approach is also mentioned as a workaround in the bug report.

I suggest adding the following in a separate `RUN` line:

```Dockerfile
# Remove the built-in `ubuntu` user and group to restore a clean state
RUN userdel -fr ubuntu && groupdel -f ubuntu
```

With this change, the rest of the code can remain unchanged.

LCerebo replied:
I've update the PR code, the 
```groupdel -f ubuntu``` is not necessary since  ```userdel -fr ubuntu``` already delete the group.

FulvioRizzardiniEng committed:
change: deleting ubuntu user before adding jenkins user.
Since ubuntu 23.04 the ubuntu user with id 1000 is present in the image, see https://bugs.launchpad.net/cloud-images/+bug/2005129.

shamil commented:
Thanks for the contribution @LCerebo !
